core/src/avm2/activation.rs

```diff
@@ -1196,17 +1196,15 @@ impl<'a, 'gc> Activation<'a, 'gc> {
     fn return_void(&mut self, return_type: Option<Class<'gc>>) -> Value<'gc> {
         // Manual coerce_to_type
 
-        let class_defs = self.avm2().class_defs();
-
         if let Some(return_type) = return_type {
-            if return_type == class_defs.void {
+            if return_type.is_builtin_void() {
                 Value::Undefined
-            } else if return_type == class_defs.int
-                || return_type == class_defs.uint
-                || return_type == class_defs.number
+            } else if return_type.is_builtin_int()
+                || return_type.is_builtin_uint()
+                || return_type.is_builtin_number()
```

Comment:
This check appears often enough that it may be worth a dedicated `.is_numeric()` method.